mne/preprocessing/bads.py

```diff
@@ -47,3 +57,75 @@ def _find_outliers(X, threshold=3.0, max_iter=2, tail=0):
 
     bad_idx = np.where(my_mask)[0]
     return bad_idx
+
+
+def unify_bad_channels(insts):
+    """Unify bad channels across list of instances.
+
+    This function looks across the list of instances to gather a list of
+    "bad" channels. Every instance's info["bads"] is set with the same "bad"
+    channels.
+
+    Parameters
+    ----------
+    insts : list
+        List of instances (:class:`~mne.io.Raw`, :class:`~mne.Epochs`,
+        :class:`~mne.Evoked`, :class:`~mne.time_frequency.Spectrum`,
+        :class:`~mne.time_frequency.EpochSpectrum`) to unify bad channels.
+
+    Returns
+    -------
+    new_insts : list
+        List of instances with bad channels unified across instances.
+
+    Notes
+    -----
+    This function operates in-place.
+
+    .. versionadded:: 1.6
+    """
+    # check input is not an empty list
+    if len(insts) == 0:
+        raise ValueError("Be sure insts is not empty list")
+
+    # first check that each object is mne object
+    inst_type = type(insts[0])
+    valid_types = (BaseRaw, Epochs, Evoked, BaseSpectrum)
+    for inst in insts:
+        _validate_type(inst, valid_types, "instance type")
+        if type(inst) != inst_type:
+            raise ValueError("All insts must be the same type")
+
+    # check that input is a list
+    if not isinstance(insts, list):
+        raise ValueError(f"insts must be a *list* of mne objects, got {type(insts)}")
```

Comment:
this goes away, replaced by an additional call to `_validate_type(..., list)` at the very beginning.